app/Http/Controllers/ChatController.php

```diff
@@ -32,16 +31,26 @@ public function index()
         $user = auth()->user();
         Chat::ack($user);
 
+        $targetId = get_params(request()->all(), null, ['sendto:int'], ['null_missing' => true])['sendto'];
+        // rejoin any existing channel first, otherwise it'll be missing from the preload later.
+        $targetUser = $targetId !== null ? User::lookup($targetId, 'id') : null;
+        if ($targetUser !== null) {
+            $channel = Channel::findPM($targetUser, $user);
+            $channel?->addUser($user);
```

Comment:
looks like the channel loaded inside `$userChannel` need to call `resetMemoized` as well in the `addUser` function.

And maybe move the `$this->resetMemoized()` (a few lines up) outside.

https://github.com/ppy/osu-web/blob/756e37cc4ea7b78fc92160f1cffc4043943e2ef5/app/Models/Chat/Channel.php#L421